hana/enable_cost_optimized.sls

```diff
@@ -1,6 +1,10 @@
 {%- from "hana/map.jinja" import hana with context -%}
 {%- from 'hana/macros/get_hana_exe_extract_dir.sls' import get_hana_exe_extract_dir with context %}
-{% set hana_extract_dir = get_hana_exe_extract_dir(hana) %}
+{%- if hana.client_archive_file is defined and hana.hana_archive_file is defined and hana.hana_archive_file.endswith((".sar", ".SAR")) %}
```

Comment:
This condition looks pretty confusing. I would add maybe a comment above to say what it does.
I understand that if hana archive is sar format, and there is a hana client archive, it uses this path. Otherwise, it uses the hana platform path

Reply:
Besides, I see this piece of code duplicated. Should we create a macro for this too?

Reply:
I am not sure about that. This code is calling the previous macro. Nested macros maybe confusing

Reply:
Why? I don't see them confusing. It's quite natural in programming (methods calling other methods).
Actually having duplicated code in pieces that we don't look often looks risky. Most probably we would forget to do the changes

Reply:
yes I tested the code with macro. And from maintenance purpose as well it makes sense to add the macro for this code